[team4_pair2] 자동차 경주 게임 1단계 구현 및 테스트 추가

Kelicia91 opened the pull request:
자동차 경주 게임 피드백 후 merge 요청드립니다.

Commit:
자동차 경주 게임 1단계 구현 및 테스트 추가

javajigi commented on src/main/java/racingcar/RacingGame.java:
```diff
@@ -0,0 +1,52 @@
+        cars = new ArrayList<>();
+    }
+
+    public void settingCars(int num) {
```
이 메소드를 생성자에서 구현하면 어떨까?

javajigi commented on src/main/java/racingcar/RacingGame.java:
```diff
@@ -0,0 +1,52 @@
+    public String repeatDash(int num) {
+        StringBuffer sb = new StringBuffer();
+        for (int i = 0; i < num; i++) {
+            sb.append("-");
```
값을 하드코딩하지 마라.
상수(static final)를 만들고 이름을 부여해 이 변수의 역할이 무엇인지 의도를 드러내라.

javajigi commented on src/main/java/racingcar/RacingGame.java:
```diff
@@ -0,0 +1,52 @@
+        }
+    }
+
+    public String repeatDash(int num) {
```
인스턴스 변수에 의존하지 않아 클래스 메소드로 구현해도 되지 않을까요?
이와 비슷한 종류의 메소드를 찾아 클래스 메소드로 구현한다.

javajigi reviewed (changes requested):
구현 잘 했네요. 피드백 남겼어요.

Commit:
* step1 review
- settingCars 메소드를 생성자로 이동
- repeatDash 메소드를 utility.StringUtility 로 분리
- "-" 상수화

minsungryu commented:
@javajigi 피드백해주신 부분 수정해서 반영했습니다!

Commit:
* del : RacingGameTest
* add : StringUtiltiyTest

minsungryu commented on src/main/java/calculator/StringCalculator.java:
```diff
@@ -1,4 +1,4 @@
-package stringCalculator;
```
step0 리팩토링하려다가 잘못 반영된 부분입니다.
